### ISSUE TITLE
Upstream Deprecation Warnings

### ISSUE THREAD

**Issue by garrettwrong** [ComputationalCryoEM/ASPIRE-Python#852]
For another task I was co-installing ASPIRE in an upstream environment and received some deprecation warnings.  Mainly they are relating to scipy namespace, and most look pretty straightforward to resolve (changing some imports).

```
../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/isolve/iterative.py:10
../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/isolve/iterative.py:10
../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/isolve/iterative.py:10
  /Users/gbwright/opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/isolve/iterative.py:10: DeprecationWarning: Please use `LinearOperator` from the `scipy.sparse.linalg` namespace, the `scipy.sparse.linalg.interface` namespace is deprecated.
    from scipy.sparse.linalg.interface import LinearOperator

../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/isolve/utils.py:7
../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/isolve/utils.py:7
../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/isolve/utils.py:7
  /Users/gbwright/opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/isolve/utils.py:7: DeprecationWarning: Please use `asmatrix` from the `scipy.sparse` namespace, the `scipy.sparse.sputils` namespace is deprecated.
    from scipy.sparse.sputils import asmatrix

../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/isolve/utils.py:9
../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/isolve/utils.py:9
../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/isolve/utils.py:9
  /Users/gbwright/opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/isolve/utils.py:9: DeprecationWarning: Please use `aslinearoperator` from the `scipy.sparse.linalg` namespace, the `scipy.sparse.linalg.interface` namespace is deprecated.
    from scipy.sparse.linalg.interface import aslinearoperator, LinearOperator, \

../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/isolve/utils.py:9
../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/isolve/utils.py:9
../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/isolve/utils.py:9
  /Users/gbwright/opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/isolve/utils.py:9: DeprecationWarning: Please use `LinearOperator` from the `scipy.sparse.linalg` namespace, the `scipy.sparse.linalg.interface` namespace is deprecated.
    from scipy.sparse.linalg.interface import aslinearoperator, LinearOperator, \

../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/isolve/utils.py:9
../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/isolve/utils.py:9
../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/isolve/utils.py:9
  /Users/gbwright/opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/isolve/utils.py:9: DeprecationWarning: Please use `IdentityOperator` from the `scipy.sparse.linalg` namespace, the `scipy.sparse.linalg.interface` namespace is deprecated.
    from scipy.sparse.linalg.interface import aslinearoperator, LinearOperator, \

../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/isolve/lsqr.py:56
../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/isolve/lsqr.py:56
../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/isolve/lsqr.py:56
  /Users/gbwright/opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/isolve/lsqr.py:56: DeprecationWarning: Please use `aslinearoperator` from the `scipy.sparse.linalg` namespace, the `scipy.sparse.linalg.interface` namespace is deprecated.
    from scipy.sparse.linalg.interface import aslinearoperator

../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/isolve/lsmr.py:24
../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/isolve/lsmr.py:24
../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/isolve/lsmr.py:24
  /Users/gbwright/opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/isolve/lsmr.py:24: DeprecationWarning: Please use `aslinearoperator` from the `scipy.sparse.linalg` namespace, the `scipy.sparse.linalg.interface` namespace is deprecated.
    from scipy.sparse.linalg.interface import aslinearoperator

../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/dsolve/linsolve.py:7
../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/dsolve/linsolve.py:7
../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/dsolve/linsolve.py:7
  /Users/gbwright/opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/dsolve/linsolve.py:7: DeprecationWarning: Please use `is_pydata_spmatrix` from the `scipy.sparse` namespace, the `scipy.sparse.sputils` namespace is deprecated.
    from scipy.sparse.sputils import is_pydata_spmatrix

../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/eigen/arpack/arpack.py:47
../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/eigen/arpack/arpack.py:47
../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/eigen/arpack/arpack.py:47
  /Users/gbwright/opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/eigen/arpack/arpack.py:47: DeprecationWarning: Please use `aslinearoperator` from the `scipy.sparse.linalg` namespace, the `scipy.sparse.linalg.interface` namespace is deprecated.
    from scipy.sparse.linalg.interface import aslinearoperator, LinearOperator

../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/eigen/arpack/arpack.py:47
../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/eigen/arpack/arpack.py:47
../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/eigen/arpack/arpack.py:47
  /Users/gbwright/opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/eigen/arpack/arpack.py:47: DeprecationWarning: Please use `LinearOperator` from the `scipy.sparse.linalg` namespace, the `scipy.sparse.linalg.interface` namespace is deprecated.
    from scipy.sparse.linalg.interface import aslinearoperator, LinearOperator

../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/eigen/arpack/arpack.py:50
../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/eigen/arpack/arpack.py:50
../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/eigen/arpack/arpack.py:50
  /Users/gbwright/opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/eigen/arpack/arpack.py:50: DeprecationWarning: Please use `isdense` from the `scipy.sparse` namespace, the `scipy.sparse.sputils` namespace is deprecated.
    from scipy.sparse.sputils import isdense, is_pydata_spmatrix

../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/eigen/arpack/arpack.py:50
../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/eigen/arpack/arpack.py:50
../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/eigen/arpack/arpack.py:50
  /Users/gbwright/opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/sparse/linalg/eigen/arpack/arpack.py:50: DeprecationWarning: Please use `is_pydata_spmatrix` from the `scipy.sparse` namespace, the `scipy.sparse.sputils` namespace is deprecated.
    from scipy.sparse.sputils import isdense, is_pydata_spmatrix

<frozen importlib._bootstrap>:219
<frozen importlib._bootstrap>:219
<frozen importlib._bootstrap>:219
  <frozen importlib._bootstrap>:219: RuntimeWarning: scipy._lib.messagestream.MessageStream size changed, may indicate binary incompatibility. Expected 56 from C header, got 64 from PyObject

../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/io/matlab/__init__.py:52
../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/io/matlab/__init__.py:52
../../../opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/io/matlab/__init__.py:52
  /Users/gbwright/opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/scipy/io/matlab/__init__.py:52: DeprecationWarning: Please use `OPAQUE_DTYPE` from the `scipy.io.matlab` namespace, the `scipy.io.matlab.mio5_params` namespace is deprecated.
    from .import (mio, mio5, mio5_params, mio4, byteordercodes,

tests/test_CtfEstimate.py::CtfEstimatorTestCase::testEstimateCTF
tests/test_CtfEstimate.py::CtfEstimatorTestCase::testRectangularMicrograph_0
tests/test_CtfEstimate.py::CtfEstimatorTestCase::testRectangularMicrograph_1
  /Users/gbwright/opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/aspire/ctf/ctf_estimator.py:362: DeprecationWarning: `method='interior-point'` is deprecated and will be removed in SciPy 1.11.0. Please use one of the HiGHS solvers (e.g. `method='highs'`) in new code.
    x = linprog(

tests/test_load_images.py: 17 warnings
tests/test_orient_sync.py: 1 warning
tests/test_relion_source.py: 1 warning
tests/test_simulation.py: 2 warnings
tests/test_starfile_stack.py: 6 warnings
  /Users/gbwright/opt/miniconda3/envs/aspire_drgn/lib/python3.8/site-packages/aspire/source/relion.py:209: FutureWarning: In a future version of pandas all arguments of StringMethods.split except for the argument 'pat' will be keyword-only.
    df[["__mrc_index", "__mrc_filename"]] = df["_rlnImageName"].str.split(
```

**Comment by garrettwrong** [comment]
Env used

[freeze.txt](https://github.com/ComputationalCryoEM/ASPIRE-Python/files/10805379/freeze.txt)

**Comment by j-c-c** [comment]
@garrettwrong Is it possible these warnings are external to ASPIRE? I'm grepping around for these name spaces and imports and don't see any of them (aside from LinearOperator which is imported properly).

**Comment by garrettwrong** [comment]
It is possible the scipy ones were fixed by another package. For those we can just check that all the imports we do match their suggestion. 

The other warnings look like legit deprecations and should be fixed.

**Comment by j-c-c** [comment]
Closed by #864.
